yarn-project/p2p/src/mem_pools/tx_pool/aztec_kv_tx_pool.ts

```diff
@@ -171,34 +193,23 @@ export class AztecKVTxPool extends (EventEmitter as new () => TypedEventEmitter<
             await this.#blockToDeletedMinedTxHash.deleteValue(originalBlock, key);
           }
         }
-
-        await this.#minedTxHashToBlock.set(key, blockHeader.globalVariables.blockNumber);
-
-        const tx = await this.getPendingTxByHash(hash);
-        if (tx) {
-          const nullifiers = tx.data.getNonEmptyNullifiers();
-          nullifiers.forEach(nullifier => minedNullifiers.add(nullifier.toString()));
-          minedFeePayers.add(tx.data.feePayer.toString());
-          pendingTxSize -= tx.getSize();
-          await this.removePendingTxIndices(tx, key);
-        }
       }
-      await this.#pendingTxSize.set(pendingTxSize);
+    });
 
-      await this.evictInvalidTxsAfterMining(txHashes, blockHeader, minedNullifiers, minedFeePayers);
+    await this.#store.transactionAsync(async () => {
```

Comment:
Why on a separate transaction? What happens if another update to the tx pool gets inbetween the marking-as-mined and the eviction?